backend/src/test/java/org/uresti/pozarreal/service/impl/HousesServiceImplTests.java

```diff
@@ -0,0 +1,223 @@
+
+    @Test
+    public void givenAnCorrectHouseId_thenReturnHouseInfo() {
+        //Given:
+        HousesRepository housesRepository = Mockito.mock(HousesRepository.class);
+        StreetRepository streetRepository = Mockito.mock(StreetRepository.class);
+        RepresentativeRepository representativeRepository = Mockito.mock(RepresentativeRepository.class);
+        SessionHelper mockSessionHelper = Mockito.mock(SessionHelper.class);
+        HousesServiceImpl housesService = new HousesServiceImpl(housesRepository, streetRepository, representativeRepository, mockSessionHelper);
+
+        List<org.uresti.pozarreal.model.House> houses = new LinkedList<>();
+
+        org.uresti.pozarreal.model.House house = new org.uresti.pozarreal.model.House();
+        house.setId("1");
+        house.setNotes("hello");
+        house.setStreet("123");
+        house.setNumber("123456");
+        house.setChipsEnabled(true);
+        houses.add(house);
+
+        Street street = new Street();
+
+        street.setId("1");
+        street.setName("name 1");
+
+        // When:
+        Mockito.when(housesRepository.findAllByStreetOrderByNumber("123")).thenReturn(houses);
+        Mockito.when(housesRepository.findById("1")).thenReturn(java.util.Optional.of(house));
+        Mockito.when(streetRepository.findById(house.getStreet())).thenReturn(java.util.Optional.of(street));
+
+        // Then:
+        HouseInfo houseInfo = housesService.getHouseInfo("1");
+
+        Assertions.assertThat(houseInfo.getNotes()).isEqualTo("hello");
+        Assertions.assertThat(houseInfo.getStreet()).isEqualTo("123");
+        Assertions.assertThat(houseInfo.getNumber()).isEqualTo("123456");
+        Assertions.assertThat(houseInfo.getStreetName()).isEqualTo("name 1");
+        Assertions.assertThat(houseInfo.getId()).isEqualTo("1");
+    }
+
+    @Test
+    public void givenStringIdCorrect_whenGetNotes_thenSaveNotes() {
+        //Given:
+        HousesRepository housesRepository = Mockito.mock(HousesRepository.class);
+        StreetRepository streetRepository = Mockito.mock(StreetRepository.class);
+        RepresentativeRepository representativeRepository = Mockito.mock(RepresentativeRepository.class);
+        SessionHelper mockSessionHelper = Mockito.mock(SessionHelper.class);
+        HousesServiceImpl housesService = new HousesServiceImpl(housesRepository, streetRepository, representativeRepository, mockSessionHelper);
+
+        List<org.uresti.pozarreal.model.House> houses = new LinkedList<>();
+
+        org.uresti.pozarreal.model.House house = new org.uresti.pozarreal.model.House();
+        house.setId("1");
+        house.setStreet("123");
+        house.setNumber("123456");
+        house.setChipsEnabled(true);
+        houses.add(house);
+
+        Street street = new Street();
+
+        street.setId("1");
+        street.setName("name 1");
+
+        // When:
+        Mockito.when(housesRepository.findAllByStreetOrderByNumber("123")).thenReturn(houses);
```

Comment:
same here with moving "Mockito.when" sentences to "Given" section and "housesService.saveNotes" to "When"

Reply:
done